docs/multicluster/architecture.md

```diff
@@ -0,0 +1,91 @@
+
+## Terminology
+
+ClusterSet is a placeholder name for a group of clusters with a high degree of mutual
+trust and shared ownership that share Services amongst themselves. Within a ClusterSet,
+Namespace sameness applies, which means all Namespaces with a given name are considered to
+be the same Namespace. The ClusterSet Custom Resource Definition(CRD) defines a ClusterSet
+including the leader and member clusters information.
+
+The ClusterClaim CRD is used to claim a cluster itself as a member of a ClusterSet with
+unique cluster ID, to claim a ClusterSet with unique ClusterSet ID.
+
+The MemberClusterAnnounce CRD declares a member cluster configuration to the leader cluster.
+
+The Common Area is an abstraction in the Antrea Multi-cluster implementation provides a storage
+interface for resource export/import that can be read/written by all member and leader clusters
+in the ClusterSet. The Common Area is implemented with a Namespace in the leader cluster
+
+## Antrea Multi-cluster Controller
+
+In a member cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for exporting resource to and importing resource from a leader
+cluster in a ClusterSet.
+
+In a leader cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for converting resources from different member clusters into one
+encapsulated resource as long as these resources have the same kind and match Namespace sameness.
+
+In ClusterSet initialization, Antrea Multi-cluster Controller in a member cluster watches
+ClusterSet, ClusterClaim, and creates a MemberClusterAnnounce in the leader cluster.
+
+For exporting resources from the member cluster, it watches ServiceExport, Service and Endpoints
+resources, encapsulates Services and Endpoints into ResourceExports according to ServiceExports,
+and writes the ResourceExports to leader cluster. For resource importing, it watches ResourceImports
+from leader cluster, and creates multi-cluster Services and Endpoints with a prefix `antrea-mc-`
+plus exported Service name, and also ServiceImports which have the same name as the Service name.
+
+In a leader cluster, for ClusterSet initialization, Antrea Multi-cluster controller watches and
+validates the ClusterSet and Clusterclaim. For resource export/import, it watches ResourceExports
+and encapsulates them into ResourceImports.
+
+## Service Export and Import
+
+<img src="assets/resource-export-import-pipeline.png" width="800" alt="Antrea Multi-cluster Resource Export/Import Pipeline">
+
+The current multi-cluster implementation supports Service discovery and Service export/import among
+member clusters. The above diagram depicts Antrea Multi-cluster resource export/import pipeline.
+
+We can look at a specific Service export/import example. Given two Kubernetes Services - `foo.ns.cluster1.local`
```

Comment:
This way might be better: "Given two Services in the member clusters - `foo.ns.cluster1.local` and `foo.ns.cluster2.local`,"

Reply:
done